File: src/lib/types/day-plan.ts
import type { ChecklistItem } from './checklist';
import type { QuizQuestion } from './quiz';
import type { ConceptTag } from './concept-tag';
import type { DayDependency } from './dependency';

export interface DayPlan {
	id: string;
	program_id: string;
	module_id: string;
	title: string;
	day_number: number;
	version: number;
	status: 'draft' | 'published' | 'archived';
	syntax_targets: string;
	implementation_brief: string;
	files_to_create: string;
	success_criteria: string;
	stretch_challenge: string;
	notes: string;
	estimated_minutes: number;
	memory_rebuild_minutes: number;
	min_minutes: number;
	recommended_minutes: number;
	deep_minutes: number;
	complexity_level: number;
	focus_blocks: string;
	created_at: string;
	updated_at: string;
}

export interface FocusBlock {
	session_type: string;
	minutes: number;
}

export interface DayPlanFull {
	day_plan: DayPlan;
	checklist_items: ChecklistItem[];
	quiz_questions: QuizQuestion[];
	concept_tags: ConceptTag[];
	dependencies: DayDependency[];
	module_title: string;
	module_color: string;
}

export interface DayPlanSummary {
	id: string;
	title: string;
	day_number: number;
	module_id: string;
	module_title: string;
	module_color: string;
	status: string;
	estimated_minutes: number;
	checklist_count: number;
	quiz_count: number;
	tag_count: number;
	best_score: number | null;
	attempt_count: number;
}

export interface CreateDayPlanInput {
	program_id: string;
	module_id: string;
	title: string;
	day_number: number;
	syntax_targets: string;
	implementation_brief: string;
	min_minutes?: number;
	recommended_minutes?: number;
	deep_minutes?: number;
	complexity_level?: number;
}

export interface UpdateDayPlanInput {
	title?: string;
	syntax_targets?: string;
	implementation_brief?: string;
	files_to_create?: string;
	success_criteria?: string;
	stretch_challenge?: string;
	notes?: string;
	status?: string;
	estimated_minutes?: number;
	memory_rebuild_minutes?: number;
}

File: src-tauri/src/db/models/day_plan.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct DayPlan {
    pub id: String,
    pub program_id: String,
    pub module_id: String,
    pub title: String,
    pub day_number: i32,
    pub version: i32,
    pub status: String,
    pub syntax_targets: String,
    pub implementation_brief: String,
    pub files_to_create: String,
    pub success_criteria: String,
    pub stretch_challenge: String,
    pub notes: String,
    pub estimated_minutes: i32,
    pub memory_rebuild_minutes: i32,
    pub min_minutes: i32,
    pub recommended_minutes: i32,
    pub deep_minutes: i32,
    pub complexity_level: i32,
    pub focus_blocks: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FocusBlock {
    pub session_type: String,
    pub minutes: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDayPlanInput {
    pub program_id: String,
    pub module_id: String,
    pub title: String,
    pub day_number: i32,
    pub syntax_targets: String,
    pub implementation_brief: String,
    pub min_minutes: Option<i32>,
    pub recommended_minutes: Option<i32>,
    pub deep_minutes: Option<i32>,
    pub complexity_level: Option<i32>,
}
